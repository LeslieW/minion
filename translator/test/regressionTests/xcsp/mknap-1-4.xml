<?xml version="1.0" encoding="UTF-8"?>

<instance>
<presentation name="?" maxConstraintArity="28" format="XCSP 2.0"/>

<domains nbDomains="1">
<domain name="D0" nbValues="2">0..1</domain>
</domains>

<variables nbVariables="28">
<variable name="V0" domain="D0"/>
<variable name="V1" domain="D0"/>
<variable name="V2" domain="D0"/>
<variable name="V3" domain="D0"/>
<variable name="V4" domain="D0"/>
<variable name="V5" domain="D0"/>
<variable name="V6" domain="D0"/>
<variable name="V7" domain="D0"/>
<variable name="V8" domain="D0"/>
<variable name="V9" domain="D0"/>
<variable name="V10" domain="D0"/>
<variable name="V11" domain="D0"/>
<variable name="V12" domain="D0"/>
<variable name="V13" domain="D0"/>
<variable name="V14" domain="D0"/>
<variable name="V15" domain="D0"/>
<variable name="V16" domain="D0"/>
<variable name="V17" domain="D0"/>
<variable name="V18" domain="D0"/>
<variable name="V19" domain="D0"/>
<variable name="V20" domain="D0"/>
<variable name="V21" domain="D0"/>
<variable name="V22" domain="D0"/>
<variable name="V23" domain="D0"/>
<variable name="V24" domain="D0"/>
<variable name="V25" domain="D0"/>
<variable name="V26" domain="D0"/>
<variable name="V27" domain="D0"/>
</variables>

<predicates nbPredicates="4">
<predicate name="P0">
  <parameters>int X0 int X1 int X2 int X3 int X4 int X5 int X6 int X7 int X8 int X9 int X10 int X11 int X12 int X13 int X14 int X15 int X16 int X17 int X18 int X19 int X20 int X21 int X22 int X23 int X24 int X25 int X26 int X27 int X28 int X29 int X30 int X31 int X32 int X33 int X34 int X35 int X36 int X37 int X38 int X39 int X40 int X41 int X42 int X43 int X44 int X45 int X46 int X47 int X48 int X49 int X50 int X51 int X52 int X53 int X54 int X55 int X56</parameters>
  <expression> 
    <functional>le(add(mul(X0,X1),add(mul(X2,X3),add(mul(X4,X5),add(mul(X6,X7),add(mul(X8,X9),add(mul(X10,X11),add(mul(X12,X13),add(mul(X14,X15),add(mul(X16,X17),add(mul(X18,X19),add(mul(X20,X21),add(mul(X22,X23),add(mul(X24,X25),add(mul(X26,X27),add(mul(X28,X29),add(mul(X30,X31),add(mul(X32,X33),add(mul(X34,X35),add(mul(X36,X37),add(mul(X38,X39),add(mul(X40,X41),add(mul(X42,X43),add(mul(X44,X45),add(mul(X46,X47),add(mul(X48,X49),add(mul(X50,X51),add(mul(X52,X53),mul(X54,X55)))))))))))))))))))))))))))),X56)</functional>
  </expression>
</predicate>
<predicate name="P1">
  <parameters>int X0 int X1 int X2 int X3 int X4 int X5 int X6 int X7 int X8 int X9 int X10 int X11 int X12 int X13 int X14 int X15 int X16 int X17 int X18 int X19 int X20 int X21 int X22 int X23 int X24 int X25 int X26 int X27 int X28 int X29 int X30 int X31 int X32 int X33 int X34 int X35 int X36 int X37 int X38 int X39 int X40 int X41 int X42 int X43 int X44 int X45 int X46 int X47 int X48 int X49 int X50 int X51 int X52 int X53 int X54 int X55</parameters>
  <expression> 
    <functional>le(add(mul(X0,X1),add(mul(X2,X3),add(mul(X4,X5),add(mul(X6,X7),add(mul(X8,X9),add(mul(X10,X11),add(mul(X12,X13),add(mul(X14,X15),add(mul(X16,X17),add(mul(X18,X19),add(mul(X20,X21),add(mul(X22,X23),add(mul(X24,X25),add(mul(X26,X27),add(mul(X28,X29),add(mul(X30,X31),add(mul(X32,X33),add(mul(X34,X35),add(X36,add(mul(X37,X38),add(mul(X39,X40),add(mul(X41,X42),add(mul(X43,X44),add(mul(X45,X46),add(mul(X47,X48),add(mul(X49,X50),add(mul(X51,X52),mul(X53,X54)))))))))))))))))))))))))))),X55)</functional>
  </expression>
</predicate>
<predicate name="P2">
  <parameters>int X0 int X1 int X2 int X3 int X4 int X5 int X6 int X7 int X8 int X9 int X10 int X11 int X12 int X13 int X14 int X15 int X16 int X17 int X18 int X19 int X20 int X21 int X22 int X23 int X24 int X25 int X26 int X27 int X28 int X29 int X30 int X31 int X32 int X33 int X34 int X35 int X36 int X37 int X38 int X39 int X40 int X41 int X42 int X43 int X44 int X45 int X46 int X47 int X48 int X49 int X50 int X51 int X52 int X53 int X54 int X55</parameters>
  <expression> 
    <functional>le(add(mul(X0,X1),add(mul(X2,X3),add(X4,add(mul(X5,X6),add(mul(X7,X8),add(mul(X9,X10),add(mul(X11,X12),add(mul(X13,X14),add(mul(X15,X16),add(mul(X17,X18),add(mul(X19,X20),add(mul(X21,X22),add(mul(X23,X24),add(mul(X25,X26),add(mul(X27,X28),add(mul(X29,X30),add(mul(X31,X32),add(mul(X33,X34),add(mul(X35,X36),add(mul(X37,X38),add(mul(X39,X40),add(mul(X41,X42),add(mul(X43,X44),add(mul(X45,X46),add(mul(X47,X48),add(mul(X49,X50),add(mul(X51,X52),mul(X53,X54)))))))))))))))))))))))))))),X55)</functional>
  </expression>
</predicate>
<predicate name="P3">
  <parameters>int X0 int X1 int X2 int X3 int X4 int X5 int X6 int X7 int X8 int X9 int X10 int X11 int X12 int X13 int X14 int X15 int X16 int X17 int X18 int X19 int X20 int X21 int X22 int X23 int X24 int X25 int X26 int X27 int X28 int X29 int X30 int X31 int X32 int X33 int X34 int X35 int X36 int X37 int X38 int X39 int X40 int X41 int X42 int X43 int X44 int X45 int X46 int X47 int X48 int X49 int X50 int X51 int X52 int X53 int X54 int X55 int X56</parameters>
  <expression> 
    <functional>eq(add(mul(X0,X1),add(mul(X2,X3),add(mul(X4,X5),add(mul(X6,X7),add(mul(X8,X9),add(mul(X10,X11),add(mul(X12,X13),add(mul(X14,X15),add(mul(X16,X17),add(mul(X18,X19),add(mul(X20,X21),add(mul(X22,X23),add(mul(X24,X25),add(mul(X26,X27),add(mul(X28,X29),add(mul(X30,X31),add(mul(X32,X33),add(mul(X34,X35),add(mul(X36,X37),add(mul(X38,X39),add(mul(X40,X41),add(mul(X42,X43),add(mul(X44,X45),add(mul(X46,X47),add(mul(X48,X49),add(mul(X50,X51),add(mul(X52,X53),mul(X54,X55)))))))))))))))))))))))))))),X56)</functional>
  </expression>
</predicate>
</predicates>

<constraints nbConstraints="11">
<constraint name="C0" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P0">
  <parameters>8 V0 24 V1 13 V2 80 V3 70 V4 80 V5 45 V6 15 V7 28 V8 90 V9 130 V10 32 V11 20 V12 120 V13 40 V14 30 V15 20 V16 6 V17 3 V18 180 V19 220 V20 50 V21 30 V22 50 V23 12 V24 5 V25 8 V26 18 V27 930</parameters>
</constraint>
<constraint name="C1" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P0">
  <parameters>8 V0 44 V1 13 V2 100 V3 100 V4 90 V5 75 V6 25 V7 28 V8 120 V9 130 V10 32 V11 40 V12 160 V13 40 V14 60 V15 55 V16 10 V17 6 V18 240 V19 290 V20 80 V21 90 V22 70 V23 27 V24 17 V25 8 V26 28 V27 1210</parameters>
</constraint>
<constraint name="C2" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P0">
  <parameters>3 V0 6 V1 4 V2 20 V3 20 V4 30 V5 8 V6 3 V7 12 V8 14 V9 40 V10 6 V11 3 V12 20 V13 5 V14 0 V15 5 V16 3 V17 0 V18 20 V19 30 V20 40 V21 10 V22 0 V23 5 V24 0 V25 0 V26 10 V27 272</parameters>
</constraint>
<constraint name="C3" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P1">
  <parameters>5 V0 9 V1 6 V2 40 V3 30 V4 40 V5 16 V6 5 V7 18 V8 24 V9 60 V10 16 V11 11 V12 30 V13 25 V14 10 V15 13 V16 5 V17 V18 80 V19 60 V20 50 V21 20 V22 30 V23 10 V24 5 V25 3 V26 20 V27 462</parameters>
</constraint>
<constraint name="C4" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P1">
  <parameters>5 V0 11 V1 7 V2 50 V3 40 V4 40 V5 19 V6 7 V7 18 V8 29 V9 70 V10 21 V11 17 V12 30 V13 25 V14 15 V15 25 V16 5 V17 V18 100 V19 70 V20 55 V21 20 V22 50 V23 15 V24 15 V25 6 V26 20 V27 532</parameters>
</constraint>
<constraint name="C5" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P0">
  <parameters>5 V0 11 V1 7 V2 55 V3 40 V4 40 V5 21 V6 9 V7 18 V8 29 V9 70 V10 21 V11 17 V12 35 V13 25 V14 20 V15 25 V16 5 V17 2 V18 110 V19 70 V20 55 V21 20 V22 50 V23 20 V24 15 V25 6 V26 20 V27 572</parameters>
</constraint>
<constraint name="C6" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P2">
  <parameters>0 V0 0 V1 V2 10 V3 4 V4 10 V5 0 V6 6 V7 0 V8 6 V9 32 V10 3 V11 0 V12 70 V13 10 V14 0 V15 0 V16 0 V17 0 V18 0 V19 30 V20 10 V21 0 V22 10 V23 10 V24 5 V25 0 V26 10 V27 240</parameters>
</constraint>
<constraint name="C7" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P1">
  <parameters>3 V0 4 V1 5 V2 20 V3 14 V4 20 V5 6 V6 12 V7 10 V8 18 V9 42 V10 9 V11 12 V12 100 V13 20 V14 5 V15 6 V16 4 V17 V18 20 V19 50 V20 30 V21 5 V22 20 V23 20 V24 10 V25 10 V26 20 V27 400</parameters>
</constraint>
<constraint name="C8" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P0">
  <parameters>3 V0 6 V1 9 V2 30 V3 29 V4 20 V5 12 V6 12 V7 10 V8 30 V9 42 V10 18 V11 18 V12 110 V13 20 V14 15 V15 18 V16 7 V17 2 V18 40 V19 60 V20 50 V21 25 V22 25 V23 25 V24 15 V25 10 V26 28 V27 470</parameters>
</constraint>
<constraint name="C9" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P0">
  <parameters>3 V0 8 V1 9 V2 35 V3 29 V4 20 V5 16 V6 15 V7 10 V8 30 V9 42 V10 20 V11 18 V12 120 V13 20 V14 20 V15 22 V16 7 V17 3 V18 50 V19 60 V20 55 V21 25 V22 30 V23 25 V24 15 V25 10 V26 28 V27 490</parameters>
</constraint>
<constraint name="C10" arity="28" scope="V0 V1 V2 V3 V4 V5 V6 V7 V8 V9 V10 V11 V12 V13 V14 V15 V16 V17 V18 V19 V20 V21 V22 V23 V24 V25 V26 V27" reference="P3">
  <parameters>100 V0 220 V1 90 V2 400 V3 300 V4 400 V5 205 V6 120 V7 160 V8 580 V9 400 V10 140 V11 100 V12 1300 V13 650 V14 320 V15 480 V16 80 V17 60 V18 2550 V19 3100 V20 1100 V21 950 V22 450 V23 300 V24 220 V25 200 V26 520 V27 12400</parameters>
</constraint>
</constraints>
</instance>
