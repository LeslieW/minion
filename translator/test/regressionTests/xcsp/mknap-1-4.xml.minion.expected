MINION 3

#This file has been automatically generated by TAILOR v0.2
#bug-reports: andrea@cs.st-and.ac.uk

# amount of common subexpressions used:0
# amount of inferred common subexpressions used:0
# amount of original variables saved (over direct equality):0
# amount of constraints:11
# amount of auxiliary variables: 0

**VARIABLES**
BOOL V0 
BOOL V1 
BOOL V2 
BOOL V3 
BOOL V4 
BOOL V5 
BOOL V6 
BOOL V7 
BOOL V8 
BOOL V9 
BOOL V10 
BOOL V11 
BOOL V12 
BOOL V13 
BOOL V14 
BOOL V15 
BOOL V16 
BOOL V17 
BOOL V18 
BOOL V19 
BOOL V20 
BOOL V21 
BOOL V22 
BOOL V23 
BOOL V24 
BOOL V25 
BOOL V26 
BOOL V27 

# aliases

**SEARCH**

PRINT [
[V0],[V1],[V2],[V3],[V4],[V5],[V6],[V7],
[V8],[V9],[V10],[V11],[V12],[V13],[V14],[V15],
[V16],[V17],[V18],[V19],[V20],[V21],[V22],[V23],
[V24],[V25],[V26],[V27]]

VARORDER [
V0,V1,V2,V3,V4,V5,V6,V7,
V8,V9,V10,V11,V12,V13,V14,V15,
V16,V17,V18,V19,V20,V21,V22,V23,
V24,V25,V26,V27]


**CONSTRAINTS**

weightedsumgeq([60,80,90,100,100,120,140,160,200,205,220,220,300,300,320,400,400,400,450,480,520,580,650,950,1100,1300,2550,3100], [V18,V17,V2,V0,V12,V7,V11,V8,V26,V6,V1,V25,V24,V4,V15,V10,V3,V5,V23,V16,V27,V9,V14,V22,V21,V13,V19,V20], 12400)
weightedsumleq([60,80,90,100,100,120,140,160,200,205,220,220,300,300,320,400,400,400,450,480,520,580,650,950,1100,1300,2550,3100], [V18,V17,V2,V0,V12,V7,V11,V8,V26,V6,V1,V25,V24,V4,V15,V10,V3,V5,V23,V16,V27,V9,V14,V22,V21,V13,V19,V20], 12400)
weightedsumleq([3,3,7,8,9,10,10,15,15,16,18,20,20,20,20,22,25,25,28,29,30,30,35,42,50,55,60,120], [V0,V18,V17,V1,V2,V26,V8,V25,V7,V6,V12,V11,V14,V15,V5,V16,V22,V24,V27,V4,V23,V9,V3,V10,V19,V21,V20,V13], 490)
weightedsumleq([2,3,6,7,9,10,10,12,12,15,15,18,18,18,20,20,25,25,25,28,29,30,30,40,42,50,60,110], [V18,V0,V1,V17,V2,V26,V8,V6,V7,V15,V25,V11,V12,V16,V14,V5,V22,V23,V24,V27,V4,V3,V9,V19,V10,V21,V20,V13], 470)
weightedsumleq([3,5,14,6,10,42,12,20,6,1,50,5,20,10,20,10,20,30,20,4,5,100,9,18,12,20,20,4], [V0,V2,V4,V6,V8,V10,V12,V14,V16,V18,V20,V22,V24,V26,V27,V25,V23,V21,V19,V17,V15,V13,V11,V9,V7,V5,V3,V1], 400)
weightedsumleq([2,5,5,6,7,9,11,15,17,18,20,20,20,20,21,21,25,25,29,35,40,40,50,55,55,70,70,110], [V18,V0,V17,V26,V2,V7,V1,V25,V12,V8,V15,V22,V24,V27,V11,V6,V14,V16,V9,V13,V4,V5,V23,V21,V3,V10,V20,V19], 572)
weightedsumleq([1,5,5,6,7,7,11,15,15,15,17,18,19,20,20,21,25,25,29,30,40,40,50,50,55,70,70,100], [V18,V0,V17,V26,V2,V7,V1,V15,V24,V25,V12,V8,V6,V22,V27,V11,V14,V16,V9,V13,V4,V5,V23,V3,V21,V10,V20,V19], 532)
weightedsumleq([1,3,5,5,5,5,6,9,10,10,11,13,16,16,18,20,20,24,25,30,30,30,40,40,50,60,60,80], [V18,V26,V0,V17,V25,V7,V2,V1,V15,V24,V12,V16,V11,V6,V8,V22,V27,V9,V14,V13,V23,V4,V3,V5,V21,V10,V20,V19], 462)
weightedsumleq([3,3,3,3,4,5,5,5,6,6,8,10,10,12,14,20,20,20,20,30,30,40,40], [V0,V12,V17,V7,V2,V14,V16,V24,V1,V11,V6,V22,V27,V8,V9,V13,V19,V3,V4,V20,V5,V10,V21], 272)
weightedsumleq([6,8,8,10,13,17,25,27,28,28,32,40,40,44,55,60,70,75,80,90,90,100,100,120,130,160,240,290], [V18,V0,V26,V17,V2,V25,V7,V24,V27,V8,V11,V12,V14,V1,V16,V15,V23,V6,V21,V22,V5,V3,V4,V9,V10,V13,V19,V20], 1210)
weightedsumleq([3,5,6,8,8,12,13,15,18,20,20,24,28,30,30,32,40,45,50,50,70,80,80,90,120,130,180,220], [V18,V25,V17,V0,V26,V24,V2,V7,V27,V12,V16,V1,V8,V15,V22,V11,V14,V6,V21,V23,V4,V3,V5,V9,V13,V10,V19,V20], 930)

**EOF**
